src/components/pages/donation_form/AddressTypeBasic.vue

```diff
@@ -1,108 +1,91 @@
-          name="addressType"
-          v-model="addressType"
-          native-value="anonymous"
-          :disabled="disableAnonymous">
-        {{ $t( 'donation_form_addresstype_option_anonymous_addresstype_basic' ) }}
-      </RadioInput>
-
-		</div>
-
-	</fieldset>
 </template>
 
-<script lang="ts">
-import { addressTypeFromName, AddressTypeModel } from '@src/view_models/AddressTypeModel';
-import { computed, defineComponent, PropType, Ref, ref, watch } from 'vue';
-import RadioInput from '@src/components/shared/legacy_form_inputs/RadioInput.vue';
+<script setup lang="ts">
 
-export default defineComponent( {
-	name: 'AddressTypeBasic',
-	components: { RadioInput },
-	props: {
-		disabledAddressTypes: Array as PropType<Array<AddressTypeModel>>,
-		initialAddressType: String,
-		isDirectDebit: Boolean,
-	},
-	setup( props, { emit } ) {
-
-		const addressType = ref( '' );
+import RadioField from '@src/components/shared/form_fields/RadioField.vue';
+import { AddressTypeModel } from '@src/view_models/AddressTypeModel';
+import { computed, ref, watch } from 'vue';
+import { useI18n } from 'vue-i18n';
 
-		const initialAddressTypeString = props.initialAddressType ? props.initialAddressType : 'unset';
+type AddressStates = 'anonymous' | 'person' | 'company' | 'unset';
 
-		switch ( addressTypeFromName( initialAddressTypeString ) ) {
-			case AddressTypeModel.ANON:
-				addressType.value = 'anonymous';
-				break;
-			case AddressTypeModel.UNSET:
-				addressType.value = initialAddressTypeString;
-				break;
-			case AddressTypeModel.PERSON:
-				addressType.value = 'person';
-				break;
-			case AddressTypeModel.COMPANY:
-				addressType.value = 'company';
-				break;
-		}
-		const type: Ref<AddressTypeModel> = ref( addressTypeFromName( initialAddressTypeString ) );
+interface Props {
+	disabledAddressTypes: AddressTypeModel[]
+	isDirectDebit: boolean;
+  addressTypeIsInvalid: boolean;
+	initialAddressType?: AddressStates;
+}
+const props = defineProps<Props>();
 
-		emit( 'set-full-selected', true );
+const emit = defineEmits( [ 'address-type', 'set-full-selected' ] );
 
-		const disableAnonymous = computed( (): boolean => props.disabledAddressTypes !== undefined && props.disabledAddressTypes.includes( AddressTypeModel.ANON ) );
+const addressType = ref<AddressStates>( props.initialAddressType ?? 'unset' );
+const disabledAddressTypesForModel = computed( () => props.disabledAddressTypes.reduce(
+	( acc, cur ) => {
+		if ( cur === AddressTypeModel.COMPANY ) {
+			acc.push( 'company' );
+		} else if ( cur === AddressTypeModel.ANON ) {
+			acc.push( 'anonymous' );
+		}
+		// Don't push person, as it is the default
+		return acc;
+	},
+	[] as AddressStates[]
+) );
+const addressTypeToAddressTypeModel = new Map<AddressStates, AddressTypeModel>( [
+	[ 'anonymous', AddressTypeModel.ANON ],
+	[ 'person', AddressTypeModel.PERSON ],
+	[ 'company', AddressTypeModel.COMPANY ],
+] );
+
+const { t } = useI18n();
+
+const addressTypeOptions = [
```

Comment:
I put these options inline in the html in the `AddressFields` component and it works pretty well. It also puts the options right where the `RadioField` is defined making it easier to see what options it has without having to go find a variable.